### PR TITLE
fix more warnings and add CI for building 3d with configure

### PR DESCRIPTION
**Pull request by WeiqunZhang**

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -179,5 +179,19 @@ jobs:
     - name: Build & Install
       run: |
         ./configure --dim 2
-        make -j2
+        make -j2 WARN_ALL=TRUE
```
Comment (ax3l) [inline .github/workflows/linux.yml:208]: do you also want to add this to the 1D case above? :)

**Review by ax3l — APPROVED** [review]
Awesome, thanks a lot!

File: .github/workflows/linux.yml (hunk 1 above)
Comment (ax3l) [reply .github/workflows/linux.yml:208]: Another idea: we can add `-Werror` in CI through XTRA flags to check PRs do not introduce warnings for selected compilers :)

File: Src/Base/AMReX.H [hunk 2]
```diff
@@ -89,6 +89,10 @@ namespace amrex
     void ExecOnFinalize (PTR_TO_VOID_FUNC);
     void ExecOnInitialize (PTR_TO_VOID_FUNC);
 
+    //! This shuts up the compiler about unused variables
+    template <class... Ts> AMREX_GPU_HOST_DEVICE
```
Comment (ax3l) [inline Src/Base/AMReX.H]: I usually also put a force inline on that one and make it a `constexpr`.
Took inspiration from boost:
- https://github.com/alpaka-group/alpaka/blob/0.5.0/include/alpaka/core/Unused.hpp

File: .github/workflows/linux.yml (hunk 1 above)
Comment (WeiqunZhang) [reply .github/workflows/linux.yml:208]: There are too many warnings in 1D we need to fix. :(  We do not fully support 1D.  So in many places we just have a function with empty body.  As for `-Werror`, we still have some warnings.  One we get rid of them, we can add that to a few CI tests.

File: Src/Base/AMReX.H (hunk 2 above)
Comment (WeiqunZhang) [reply Src/Base/AMReX.H]: Sounds good.  Will update.

Comment (WeiqunZhang) [reply Src/Base/AMReX.H]: Looks like we have take `constexpr`, because conexpr void is not supported in C++11.

Comment (ax3l) [reply Src/Base/AMReX.H]: Ah right, in C++11 mode I used a const fallback originally: https://github.com/alpaka-group/alpaka/pull/582

**Review by atmyers — APPROVED** [review]

**Pull request merged**